### DISCUSSION TITLE
Some tiles cannot be read. The server returns 204 with no content

### DISCUSSION THREAD

**Discussion by fandy00123** [maplibre/martin#1043] [Q&A]
Hi,
When I use martin(v0.11.2) as tile sverer in windows 10, `martin-main\tests\fixtures\mbtiles\world_cities.mbtiles` as datasource,  use `martin world_cicties.mbtiles` command, I found some tiles cannot be read.

* `http://127.0.0.1:3000/world_cities/5/6/19` server return 204 No Content
* `http://127.0.0.1:3000/world_cities/5/9/19` server return 200 can download a tile file

I use my mbitiles only zxy000 can read, I modified the zxy of the tiles to 000 and they can be read
I use other vector tile datasouce also. in my mbtiles

Looking forward to your answer!

[tiles.zip](https://github.com/maplibre/martin/files/13594586/tiles.zip)

**Comment by nyurik** [comment]
This is by design, the `5/6/19` does not exist in that file. If you inspect the file (i did it on linux, so adjust the file path):

```bash
sqlite3 tests/fixtures/mbtiles/world_cities.mbtiles 'select * from tiles where zoom_level=5 and tile_column=6'
5|6|19|
```

You see the `5|6|19` but the row (Y value) is NOT 19 because MBTiles uses inverted Y values. To get the right value, use `(2 ^ zoom) - 1 - Y`, so for this case its `2^5-1-19 = 12`.   Try getting `http://127.0.0.1:3000/world_cities/5/6/12` and it should work.
